src/Platform/Microsoft.Testing.Platform/OutputDevice/Terminal/NonAnsiTerminal.cs

```diff
@@ -90,8 +91,11 @@ public void StartUpdate()
         }
 
         bool lockTaken = false;
-        // SystemConsole.ConsoleOut is set only once in static ctor.
-        // So we are sure we will be doing Monitor.Exit on the same instance.
+
+        // We store Console.Out in a field to make sure we will be doing
+        // the Monitor.Exit call on the same instance.
+        _batchingLock = Console.Out;
```

Comment:
as you described https://source.dot.net/#System.Console/System/ConsolePal.Unix.cs,898f4da5a113560c console pal on unix is locking on console.out (so on the object that someone - like TUnit substituted), it would be nice to add that reasoning somewhere so we know what are the rules we need to follow when locking on output. To prevent deadlock and preventing mixed output from our internal services (that use our SystemConsole class) and test framework or other test code that substituted Console.Out.

And also that we know we can stop using that pattern if the underlying implementation changes.